publisher/publisher.go

```diff
@@ -87,7 +91,15 @@ func (pub *Impl) SubmitToCT(der []byte) error {
 
 	chain := append([]ct.ASN1Cert{der}, pub.issuerBundle...)
 	for _, ctLog := range pub.ctLogs {
-		sct, err := ctLog.client.AddChain(chain)
+		var sct *ct.SignedCertificateTimestamp
+		var err error
+		if pub.submissionTimeout > 0 {
+			ctx, cancel := context.WithTimeout(context.Background(), pub.submissionTimeout)
+			defer cancel()
+			sct, err = ctLog.client.AddChainWithContext(ctx, chain)
+		} else {
+			sct, err = ctLog.client.AddChain(chain)
```

Comment:
Alternately: Do the defaulting in the constructor instead, and set the timeout very high (like a day) if it is not provided. That way, when we switch it to be a required value, the logic to fail will still be in the same place.